lib/linux_admin/ssh.rb

```diff
@@ -0,0 +1,96 @@
+module LinuxAdmin
+  class SSH
+    attr_reader :ip
+    attr_reader :username
+    attr_reader :private_key
+    attr_reader :agent
+
+    def initialize(ip, username, private_key)
+      @ip = ip
+      @private_key = private_key
+      @username = username
+    end
+
+    def perform_commands(commands = [], options = {})
+      result = nil
```

Comment:
I don't think you need this result variable in the method.  Nothing happens after running the commands.